Common/cpp/LayoutAnimations/LayoutAnimationsManager.cpp

```diff
@@ -112,6 +120,37 @@ int LayoutAnimationsManager::findPrecedingViewTagForTransition(int tag) {
   return -1;
 }
 
+#ifdef DEBUG
+std::string LayoutAnimationsManager::getScreenSharedTagPairString(
+    const int screenTag,
+    const std::string &sharedTag) {
+  return std::to_string(screenTag) + ":" + sharedTag;
+}
+
+void LayoutAnimationsManager::checkDuplicateSharedTag(
+    const int viewTag,
+    const int screenTag) {
+  if (!viewTagToSharedTag_.count(viewTag)) {
+    return;
+  }
+  const auto &sharedTag = viewTagToSharedTag_[viewTag];
+  const auto &pair = getScreenSharedTagPairString(screenTag, sharedTag);
+  bool hasDuplicate = screenSharedTagSet_.count(pair);
+  if (hasDuplicate) {
+    jsLogger_->warnOnJS(
```

Comment:
Since `jsLogger_` is set later on, should we add an assert here?
```suggestion
    assert(jsLogger_ != nullptr);
    jsLogger_->warnOnJS(
```

Reply:
Is it set later on?